### PR TITLE
✨feat(front) PLAS-15: Text Components in Design System

### PR DESCRIPTION
**Pull request by bvelitchkine**
# Context

In the process of reproducing the original extension, we might need some text paragraphs here and there.

# Solution

Three different types of text paragraphs. The only thing that changes is the **font size**.

### Side Note
You'll see some `<em>` tags here and there: it's just to make sure that HTML could properly be passed as children, and not solely strings.

### CODE REVIEW

**Comment by bvelitchkine** [comment]
PS : I've added a variety of labels to the project. I think we could remove "`type`(`scope`)" from the titles of the pull requests as long as we put the two corresponding labels on them. **What do you think?**

In purple, the types of pull requests:
- fix
- feature
- improvement

In green, the possible scopes:
- docs
- infra
- design
- front
- back

**Comment by GauvainThery** [comment]
> PS : I've added a variety of labels to the project. I think we could remove "`type`(`scope`)" from the titles of the pull requests as long as we put the two corresponding labels on them. **What do you think?**
> 
> In purple, the types of pull requests:
> 
> * fix
> * feature
> * improvement
> 
> In green, the possible scopes:
> 
> * docs
> * infra
> * design
> * front
> * back

The purpose of naming the PR was to have this name in the commit (even though I haven't found how to name the merge commit with the PR name) and (in the future) trigger different workflow based on the scope of the PR - not sure this is possible with tags 😬

File: libs/design-system/src/atoms/typography/baseText.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  children: React.ReactNode;
+};
+
+const fontSizeMapping = {
+  sm: 'text-3.5',
+  base: 'text-4',
+  lg: 'text-5',
+};
+
+const aligmentMapping = {
```
Comment (GauvainThery) [inline libs/design-system/src/atoms/typography/baseText.tsx]: `alignmentMapping`

File: libs/design-system/src/atoms/typography/baseText.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  left: 'text-left',
+  center: 'text-center',
+  right: 'text-right',
+};
+
+const Paragraph = ({ type = 'base', alignment = 'left', className, children }: ParagraphProps) => {
+  return (
+    <p className={cn('antialiased', 'text-gray', fontSizeMapping[type], aligmentMapping[alignment], className)}>
```
Comment (GauvainThery) [inline libs/design-system/src/atoms/typography/baseText.tsx]: `alignmentMapping`

You can have a look at the VSCode extension Code Spell Checker

**Review by GauvainThery — APPROVED** [review]
Just a typo and the rest looks good 🔥 

As you can see the linter failed in the GitHub workflow. This is because of a rule included in the react-eslint plugin ([react/no-unescaped-entities](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unescaped-entities.md)). 
Honestly, we don't care about this rule. You can add this line in `rules` of the `.eslintrc.cjs` of the design-system to get ride of it: `'react/no-unescaped-entities': 'off',`

**Comment by bvelitchkine** [comment]
> > PS : I've added a variety of labels to the project. I think we could remove "`type`(`scope`)" from the titles of the pull requests as long as we put the two corresponding labels on them. **What do you think?**
> > In purple, the types of pull requests:
> > 
> > * fix
> > * feature
> > * improvement
> > 
> > In green, the possible scopes:
> > 
> > * docs
> > * infra
> > * design
> > * front
> > * back
> 
> The purpose of naming the PR was to have this name in the commit (even though I haven't found how to name the merge commit with the PR name) and (in the future) trigger different workflow based on the scope of the PR - not sure this is possible with tags 😬

Yeah got it, now I remember. Would be cool if it was though!

**Pull request merged**